homeassistant/components/tesla_fleet/helpers.py

```diff
@@ -0,0 +1,79 @@
+"""Tesla Fleet helper functions."""
+
+import asyncio
+from typing import Any
+
+from tesla_fleet_api.exceptions import TeslaFleetError
+
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, LOGGER, TeslaFleetState
+from .models import TeslaFleetVehicleData
+
+
+async def wake_up_vehicle(vehicle: TeslaFleetVehicleData) -> None:
+    """Wake up a vehicle."""
+    async with vehicle.wakelock:
+        times = 0
+        while vehicle.coordinator.data["state"] != TeslaFleetState.ONLINE:
+            try:
+                if times == 0:
+                    cmd = await vehicle.api.wake_up()
+                else:
+                    cmd = await vehicle.api.vehicle()
+                state = cmd["response"]["state"]
+            except TeslaFleetError as e:
+                raise HomeAssistantError(str(e)) from e
+            vehicle.coordinator.data["state"] = state
+            if state != TeslaFleetState.ONLINE:
+                times += 1
+                if times >= 4:  # Give up after 30 seconds total
+                    raise HomeAssistantError("Could not wake up vehicle")
+                await asyncio.sleep(times * 5)
+
+
+async def handle_command(command) -> dict[str, Any]:
```

Comment:
My impression is the current design pattern is to not have half-way typed arguments, so type these parameters since the return values are also typed here and below.

Reply:
Added `Awaitable`